libs/libc/modlib/modlib_registry.c

```diff
@@ -82,40 +68,7 @@ static FAR struct module_s *g_mod_registry;
 
 void modlib_registry_lock(void)
 {
-  pid_t me;
-  int ret;
-
-  /* Do we already hold the semaphore? */
-
-  me = getpid();
-  if (me == g_modlock.holder)
-    {
-      /* Yes... just increment the count */
-
-      g_modlock.count++;
-      DEBUGASSERT(g_modlock.count > 0);
-    }
-
-  /* Take the semaphore (perhaps waiting) */
-
-  else
-    {
-      while ((ret = _SEM_WAIT(&g_modlock.lock)) < 0)
-        {
-          /* The only case that an error should occur here is if
-           * the wait was awakened by a signal.
-           */
-
-          DEBUGASSERT(_SEM_ERRNO(ret) == EINTR ||
-                      _SEM_ERRNO(ret) == ECANCELED);
-          UNUSED(ret);
-        }
-
-      /* No we hold the semaphore */
-
-      g_modlock.holder = me;
-      g_modlock.count  = 1;
-    }
+  nxrmutex_lock(&g_modlock);
```

Comment:
most likely we need to
```suggestion
  while ((ret = nxrmutex_lock(&g_modlock)) < 0)
  {
    DEBUGASSERT(ret == -ECANCELED);
  }
```